filebeat/harvester/reader/line.go

```diff
@@ -24,55 +24,34 @@ type Line struct {
 	decoder    transform.Transformer
 }
 
-func NewLine(
-	input io.Reader,
-	codec encoding.Encoding,
-	bufferSize int,
-) (*Line, error) {
-	l := &Line{}
-
-	if err := l.init(input, codec, bufferSize); err != nil {
-		return nil, err
-	}
+// NewLine creates a new Line reader object
+func NewLine(input io.Reader, codec encoding.Encoding, bufferSize int) (*Line, error) {
 
-	return l, nil
-}
+	encoder := codec.NewEncoder()
 
-func (l *Line) init(
-	reader io.Reader,
-	codec encoding.Encoding,
-	bufferSize int,
-) error {
-	l.reader = reader
-	l.codec = codec
-	l.bufferSize = bufferSize
-
-	l.codec.NewEncoder()
-	nl, _, err := transform.Bytes(l.codec.NewEncoder(), []byte{'\n'})
+	// Create newline char based on encoding
+	nl, _, err := transform.Bytes(encoder, []byte{'\n'})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	l.nl = nl
-	l.decoder = l.codec.NewDecoder()
-	l.inBuffer = streambuf.New(nil)
-	l.outBuffer = streambuf.New(nil)
-	return nil
+	return &Line{
+		reader:     input,
+		codec:      codec,
+		bufferSize: bufferSize,
+		nl:         nl,
+		decoder:    codec.NewDecoder(),
+		inBuffer:   streambuf.New(nil),
+		outBuffer:  streambuf.New(nil),
+	}, nil
 }
 
 func (l *Line) Next() ([]byte, int, error) {
-	for {
-		// read next 'potential' line from input buffer/reader
-		err := l.advance()
-		if err != nil {
-			return nil, 0, err
-		}
 
-		// check last decoded byte really being '\n'
-		buf := l.outBuffer.Bytes()
-		if buf[len(buf)-1] == '\n' {
```

Comment:
this check+loop is here in case of l.advance() did find a sub-match in input string, which is no newline character if encoding has been applied.